spec/system/users_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'rails_helper'
+
+RSpec.describe "Users", type: :system do
+  let(:user){ create(:user) }
+
+  describe 'ログイン前' do
+    describe 'ユーザー新規登録' do
+      context 'フォームの入力値が正常' do
+        it 'ユーザーの新規作成が成功する' do
+          visit sign_up_path
+          fill_in 'Email', with: 'a@example.com'
+          fill_in 'Password', with: "12345678"
+          fill_in 'Password confirmation', with: "12345678"
+          click_button 'SignUp'
+          expect(current_path).to eq login_path
+          expect(page).to have_content('User was successfully created')
+        end
+      end
+
+      context 'メールアドレスが未入力' do
+        it 'ユーザーの新規作成が失敗する' do
+          visit sign_up_path
+          fill_in 'Email', with: ''
+          fill_in 'Password', with: "12345678"
+          fill_in 'Password confirmation', with: "12345678"
+          click_button 'SignUp'
+          expect(current_path).to eq users_path
+          expect(page).to have_content('1 error prohibited this user from being saved')
+          expect(page).to have_content("Email can't be blank")
+        end
+      end
+
+      context '登録済のメールアドレスを使用' do
+        it 'ユーザーの新規作成が失敗する' do
+          visit sign_up_path
+          fill_in 'Email', with: user.email
+          fill_in 'Password', with: "12345678"
+          fill_in 'Password confirmation', with: "12345678"
+          click_button 'SignUp'
+          expect(current_path).to eq users_path
+          expect(page).to have_content('1 error prohibited this user from being saved')
+          expect(page).to have_content("Email has already been taken")
+          expect(page).to have_no_content user.email
+        end
+      end
+    end
+
+    describe 'マイページ' do
+      context 'ログインしていない状態' do
+        it 'マイページへのアクセスが失敗する' do
+          visit user_path(user)
+          expect(current_path).to eq login_path
+          expect(page).to have_content('Login required')
+        end
+      end
+    end
+  end
+
+  describe 'ログイン後' do
+    let(:user_a){ create(:user) }
+    let(:user_b){ create(:user) }
+    let(:task_a){ create(:task, user: user_a) }
```

Comment:
letに定義する変数名は意味の分かるものにしましょう！
現状では作成されるユーザーがどんなデータかが分からないです。

特に意味がなく、一般的なデータであればuserとそのまま書いてしまいましょう。
```suggestion
    let(:user){ create(:user) }
    let(:another_user){ create(:user) }
    let(:task){ create(:task, user: user) }
```